scripts/json2wal/main.go

```diff
@@ -0,0 +1,75 @@
+/*
+	json2wal converts JSON file to binary WAL file.
+
+	Usage:
+			json2wal <path-to-JSON>  <path-to-wal>
+*/
+
+package main
+
+import (
+	"github.com/tendermint/go-amino"
+	"github.com/tendermint/tendermint/types"
+	cs "github.com/tendermint/tendermint/consensus"
+	"fmt"
+	"os"
+	"io"
+	"bufio"
+	"strings"
+)
+
+var cdc = amino.NewCodec()
+
+func init() {
+	cs.RegisterConsensusMessages(cdc)
+	cs.RegisterWALMessages(cdc)
+	types.RegisterBlockAmino(cdc)
+}
+
+
+func main() {
+	if len(os.Args) < 3 {
+		fmt.Println("missing arguments: Usage:json2wal <path-to-JSON>  <path-to-wal>")
+		os.Exit(1)
+	}
+
+	f, err := os.Open(os.Args[1])
+	if err != nil {
+		panic(fmt.Errorf("failed to open WAL file: %v", err))
+	}
+	defer f.Close()
+
+	walFile, err := os.OpenFile(os.Args[2],os.O_APPEND|os.O_WRONLY|os.O_CREATE,0666)
+	if err != nil {
+		panic(fmt.Errorf("failed to open WAL file: %v", err))
+	}
+	defer walFile.Close()
+
+	br := bufio.NewReader(f)
+	dec := cs.NewWALEncoder(walFile)
+
+	for {
+		msgJson, _, err := br.ReadLine()
+		if err == io.EOF{
+			break
+		}else if err != nil {
+			panic(fmt.Errorf("failed to read file: %v", err))
+		}
+		// ignore the ENDHEIGHT in json.File
+		if strings.HasPrefix(string(msgJson),"ENDHEIGHT"){
```

Comment:
why do we ignore it? it should be present in the output file, no?

Reply:
I found the following code in the wal2json.file
```
if err == nil {
			if end, ok := msg.Msg.(cs.EndHeightMessage); ok {
				_, err = os.Stdout.Write([]byte(fmt.Sprintf("ENDHEIGHT %d\n", end.Height))) // nolint: errcheck, gas
			}
		}
```
so I ignore the "ENDHEIGHT"

Reply:
Have you tried converting existing wal to json, converting json to wal and running it? Does it work?

It should be more like:

```go
if strings.HasPrefix(string(msgJson),"ENDHEIGHT"){
h := extractHeight(string(msgJson))
err = dec.Encode(&cs.EndHeightMessage{h})
		if err != nil{
			panic(fmt.Errorf("failed to encode msg: %v", err))
		}
}
```

Reply:
Yeah, I use 'md5sum' command to check the hash of existing wal and generated wal on ubuntu,and it matched.I will upload my wal right now.

Reply:
If there are EndHeightMessage in json file， go-amino will help us  write it into generated wal.
I just ignore the prefix of 'ENDHEIGHT',not 
```
{"time":"2018-08-08T11:28:00.372359633Z","msg":{"type":"tendermint/wal/EndHeightMessage","value":{"height":"2"}}}

```

Reply:
generated json file slice
```
{"time":"2018-08-08T11:28:00.285064707Z","msg":{"type":"tendermint/wal/EventDataRoundState","value":{"height":"2","round":"0","step":"RoundStepCommit"}}}
{"time":"2018-08-08T11:28:00.372359633Z","msg":{"type":"tendermint/wal/EndHeightMessage","value":{"height":"2"}}}
ENDHEIGHT 2
```

Reply:
Oh, right. Awesome. Thank you